core/src/main/scala/org/apache/spark/scheduler/LiveListenerBus.scala

```diff
@@ -111,6 +112,15 @@ private[spark] class LiveListenerBus(conf: SparkConf) extends SparkListenerBus {
     }
   }
 
+  override protected def createTimer(listener: SparkListenerInterface): Option[Timer] = {
+    if (listener.getClass.getName.startsWith("org.apache.spark")) {
```

Comment:
Why creating listener just for "spark" listener ? We may want timings even for "third-party" listeners. It is even more important in my mind, for these listeners because they can be much less optimized and so bring a huge performance  penalty